### DISCUSSION TITLE
Higher sample rate output after demod

### DISCUSSION THREAD

**Discussion by aaknitt** [rtl-airband/RTLSDR-Airband#363] [Q&A] [answered]
I know this is an odd request that doesn't fit a 'normal' use case, but I'm wondering if there's a way to get a higher sample rate output after demodulation.  

I've tried changing the WAVE_RATE #define value from 16000 to larger values but I think doing that has some unintended consequences...the audio out via UDP had skips and misses and the levels seemed lower than normal too.  

Ideally I'd like to get the output up up to 1Msps or higher.  I tried that rate and couldn't get any UDP output due to buffer overflow errors.  I was able to get UDP output at 100ksps but with the issues mentioned above.  

For context, I'm interested in playing around with multilateration but that requires high sample rates to get sufficient time resolution.

Wondering how deep into source modification I'd need to go to make something like this work.

**Comment by charlie-foxtrot** [comment, marked as answer]
Not really sure what you're trying to do but it may be incompatible with the FFT based approach that `rtl_airband` uses.

The first stage uses an FFT to process the raw incoming samples.  This is a computation heavy process but is the only thing that is done at the raw sample rate.  The remaining processing uses single FFT bin's worth of data for each channel.  This turns out to be a very efficient approach to support multiple channels coming from a single SDR as the FFT processing is a "fixed cost" and the processing required for each additional channel is at a much lower data rate.

This FFT approach also acts a bit like a bandpass filter in that it only selects a frequency range.

The default FFT size is 512 and it is configurable, but the smallest value allowed by the code is 256.

If you want a very high output sample rate you will need to decrease the FFT size, but doing so will widen the bandwidth going into the demodulator and may bring in more noise.

A better approach for what you're doing may be to replace the FFT with a bandpass filter running at the raw sample rate.

With all this said, the output of the demodulator will be limited by the bandwidth of the signal you are demodulating.  Are you going after a signal that is 1 MHz wide?

**Reply by aaknitt** [reply]
That's clever, and I agree it probably means this won't be a good match for what I'm trying to do.  I'll play around a bit with GNUradio and see if I can come up with something.  The signal isn't 1 MHz wide, but the basic idea is to inject 1 PPS pulses from a GPS receiver into the RTL-SDR front end at the end of a transmission in order to get a high-precision timestamp of the start of the transmission.  To get adequate distance resolution for mlat the sampling rate needs to be much higher than the channel bandwidth.

**Reply by donwade** [reply]
I’m confused. “A high precision timestamp of the start of transmission” are you trying to get accurate voice timestamps or are you bringing in GPS (injecting ?)to do some funky phased  array tracking with multi receivers?

**Comment by jlpoolen** [comment]
> This turns out to be a very efficient approach to support multiple channels...
> A better approach for what you're doing may be to replace the FFT with a bandpass filter running at the raw sample rate.

So if you are focused only on a single channel, i.e. frequency, e.g. 119.1 Mhz, an ATC frequency,  the Fourier transformation is really superfluous?  I'm investigating AI transcription and learning that 8 Hz may be affecting the quality of the results, so I'm considering raising the final sampling.
